pykafka/utils/compression.py

```diff
@@ -113,7 +120,7 @@ def decode_snappy(buff):
         raise ImportError("Please install python-snappy")
     if _detect_xerial_stream(buff):
         out = StringIO()
-        body = buffer(buff[16:])
+        body = bytes(buffer(buff[16:]))
```

Comment:
Re unicode/bytestring acceptance, I think this is a case in point - I'd be in favour of fixing the test instead, so that it sends a bytearray, or bytes, or both.